projects/rocprofiler-sdk/source/lib/python/rocpd/csv.py

```diff
@@ -23,14 +23,500 @@
+            json_extract(extdata, '$.workgroup_max_size') AS Workgroup_Max_Size,
+            json_extract(extdata, '$.family_id') AS Family_Id,
+            json_extract(extdata, '$.grid_max_size') AS Grid_Max_Size,
+            json_extract(extdata, '$.local_mem_size') AS Local_Mem_Size,
+            json_extract(extdata, '$.hive_id') AS Hive_Id,
+            json_extract(extdata, '$.gpu_id') AS Gpu_Id,
+            json_extract(extdata, '$.workgroup_max_dim.x') AS Workgroup_Max_Dim_X,
+            json_extract(extdata, '$.workgroup_max_dim.y') AS Workgroup_Max_Dim_Y,
+            json_extract(extdata, '$.workgroup_max_dim.z') AS Workgroup_Max_Dim_Z,
+            json_extract(extdata, '$.grid_max_dim.x') AS Grid_Max_Dim_X,
+            json_extract(extdata, '$.grid_max_dim.y') AS Grid_Max_Dim_Y,
+            json_extract(extdata, '$.grid_max_dim.z') AS Grid_Max_Dim_Z,
+            name AS Name,
+            json_extract(extdata, '$.vendor_name') AS Vendor_Name,
+            json_extract(extdata, '$.product_name') AS Product_Name,
+            model_name AS Model_Name
+        FROM "rocpd_info_agent"
+    """
+    write_sql_query_to_csv(
+        importData, query, config.output_path, config.output_file, "agent_info", ""
+    )
+
+
+def write_kernel_csv(importData, config) -> None:
+
+    if config.agent_index_value == libpyrocpd.agent_indexing.node:  # absolute
+        agent_id = "'Agent ' || agent_abs_index"
+    elif (
+        config.agent_index_value == libpyrocpd.agent_indexing.logical_node
+    ):  # relative (default)
+        agent_id = "'Agent ' || agent_log_index"
+    elif (
+        config.agent_index_value == libpyrocpd.agent_indexing.logical_node_type
+    ):  # type-relative
+        agent_id = "agent_type || ' ' || agent_type_index"
+    else:
+        agent_id = ""
+
+    if config.kernel_rename:
+        kernel_name = "region"
+    else:
+        kernel_name = "name"
+
+    query = f"""
+        SELECT
+            guid AS Guid,
+            'KERNEL_DISPATCH' AS Kind,
+            {agent_id} AS Agent_Id,
+            queue_id AS Queue_Id,
+            stream_id AS Stream_Id,
+            tid AS Thread_Id,
+            dispatch_id AS Dispatch_Id,
+            kernel_Id AS Kernel_Id,
+            {kernel_name} AS Kernel_Name,
+            stack_id AS Correlation_Id,
+            start AS Start_Timestamp,
+            end AS End_Timestamp,
+            lds_size AS LDS_Block_Size,
+            scratch_size AS Scratch_Size,
+            vgpr_count AS VGPR_Count,
+            accum_vgpr_count AS Accum_VGPR_Count,
+            sgpr_count AS SGPR_Count,
+            workgroup_x AS Workgroup_Size_X,
+            workgroup_y AS Workgroup_Size_Y,
+            workgroup_z AS Workgroup_Size_Z,
+            grid_x AS Grid_Size_X,
+            grid_y AS Grid_Size_Y,
+            grid_z AS Grid_Size_Z
+        FROM "kernels"
+        ORDER BY
+            start ASC, end DESC
```

Comment:
```suggestion
            guid ASC, start ASC, end DESC
```
When using multiple databases, the CSVs sorted by time, get intertwined with each other.  It also differs in output from the previous .cpp implementation.  If we sort by GUID ASC first, that should make it match the previous .cpp outputs.

Reply:
LGTM now.